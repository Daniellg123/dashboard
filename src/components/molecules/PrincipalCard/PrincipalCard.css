.card {
    box-sizing: border-box;

    /* Auto layout */
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 3px 12px;
    gap: 20px;

    position: static;
    width: 195px;
    height: 140px;

    /* Background */
    background: #FAFCFF;

    border: 1px solid;
}

.content {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    gap: 15px;
}

.info {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}