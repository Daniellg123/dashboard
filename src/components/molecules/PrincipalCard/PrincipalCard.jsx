import PropTypes from "prop-types";
import { Card, Typography, Divider } from "@mui/material";
import ImageCard from "../../atoms/ImageCard/ImageCard";

import './PrincipalCard.css';

/**
 * Componente para la información principal
 */

function PrincipalCard({ img, title, units, users, selected }) {
    return (
        <Card className="card" sx={selected ? {borderColor:'#ED5E2E'} : { borderColor:'#ABB0B9' }}>
            <div className="content">
                <ImageCard imgDir={img} size={48} />
                <Typography variant="body2">{title}</Typography>
            </div>
            <Divider variant="middle"/>
            <div className="content">
                <div className="info">
                    <Typography variant="body2">{units}</Typography>
                    <Typography variant="body1">Units</Typography>
                </div>
                <Divider orientation="vertical" />
                <div className="info">
                    <Typography variant="body2">{users}</Typography>
                    <Typography variant="body1">Users</Typography>
                </div>
            </div>
        </Card>
    );
};

PrincipalCard.propTypes = {
    /**
     * Ruta de la imagen
     */
    img: PropTypes.string, 
    /**
     * Titulo del card
     */
    title: PropTypes.string,
    /**
     * Número de units
     */
    units: PropTypes.number, 
    /**
     * Número de users
     */
    users: PropTypes.number,
    /**
     * Booleano para saber si está seleccionado
     */
    selected: PropTypes.bool
};

export default PrincipalCard;