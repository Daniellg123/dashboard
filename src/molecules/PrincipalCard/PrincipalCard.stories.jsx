import PrincipalCard from "./PrincipalCard";

const routeImg = './img/Imageofemployee.png';

export default {
    title: 'Molecules/PrincipalCard',
    component: PrincipalCard,
};

const Template = (args) => <PrincipalCard {...args}/>

export const Primary = Template.bind({});
Primary.args = { title: 'Desarrollo', img: routeImg, units: '130', users: '325' };