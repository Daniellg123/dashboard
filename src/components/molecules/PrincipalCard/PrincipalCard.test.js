import { render } from '@testing-library/react';
import PrincipalCard from './PrincipalCard';

const props = {
    img: './img/logo1.png',
    title: 'Titulo',
    units: 213,
    users: 20,
    selected: false
};

it('renders correctrly', () => {
    const component = render(<PrincipalCard {...props} />)
    expect(component).toMatchSnapshot();
});

it('Title is equal to title props', () => {
    const component = render(<PrincipalCard {...props} title={'Daniel'} />)
    component.getByText('Daniel');
});

it('renders correctrly when is not selected', () => {
    const component = render(<PrincipalCard {...props} selected={false} />)
    expect(component).toMatchSnapshot();
});