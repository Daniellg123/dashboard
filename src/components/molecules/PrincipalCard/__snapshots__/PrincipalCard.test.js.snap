// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctrly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-1hs4s8w-MuiPaper-root-MuiCard-root"
      >
        <div
          class="content"
        >
          <img
            alt="./img/logo1.png"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img img css-14w8qym-MuiCardMedia-root"
            src="./img/logo1.png"
          />
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Titulo
          </p>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
        />
        <div
          class="content"
        >
          <div
            class="info"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              213
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Units
            </p>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical css-ftc7wc-MuiDivider-root"
          />
          <div
            class="info"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              20
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Users
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-1hs4s8w-MuiPaper-root-MuiCard-root"
    >
      <div
        class="content"
      >
        <img
          alt="./img/logo1.png"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img img css-14w8qym-MuiCardMedia-root"
          src="./img/logo1.png"
        />
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          Titulo
        </p>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
      />
      <div
        class="content"
      >
        <div
          class="info"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            213
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Units
          </p>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical css-ftc7wc-MuiDivider-root"
        />
        <div
          class="info"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            20
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Users
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders correctrly when is not selected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-1hs4s8w-MuiPaper-root-MuiCard-root"
      >
        <div
          class="content"
        >
          <img
            alt="./img/logo1.png"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img img css-14w8qym-MuiCardMedia-root"
            src="./img/logo1.png"
          />
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Titulo
          </p>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
        />
        <div
          class="content"
        >
          <div
            class="info"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              213
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Units
            </p>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical css-ftc7wc-MuiDivider-root"
          />
          <div
            class="info"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              20
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Users
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-1hs4s8w-MuiPaper-root-MuiCard-root"
    >
      <div
        class="content"
      >
        <img
          alt="./img/logo1.png"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img img css-14w8qym-MuiCardMedia-root"
          src="./img/logo1.png"
        />
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          Titulo
        </p>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
      />
      <div
        class="content"
      >
        <div
          class="info"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            213
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Units
          </p>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical css-ftc7wc-MuiDivider-root"
        />
        <div
          class="info"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            20
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Users
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
